app/src/main/java/io/gnosis/safe/ui/StartActivity.kt

```diff
@@ -32,24 +34,33 @@ import pm.gnosis.utils.asEthereumAddress
 import pm.gnosis.utils.asEthereumAddressString
 import javax.inject.Inject
 
-class StartActivity : BaseActivity(), SafeOverviewNavigationHandler {
+class StartActivity : BaseActivity(), SafeOverviewNavigationHandler, AppStateListener {
 
     @Inject
     lateinit var safeRepository: SafeRepository
 
     @Inject
-    lateinit var creadentialsRepository: CredentialsRepository
+    lateinit var credentialsRepository: CredentialsRepository
 
     private val toolbarBinding by lazy {
         ToolbarSafeOverviewBinding.bind(findViewById(R.id.toolbar_container))
     }
     private val toolbar by lazy { findViewById<View>(R.id.toolbar) }
     private val navBar by lazy { findViewById<BottomNavigationView>(R.id.nav_bar) }
 
+    var comingFromBackground = false
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_start)
 
+        if (settingsHandler.requirePasscodeToOpen && settingsHandler.usePasscode) {
```

Comment:
could you move it to onResume and extend then when statement there? Navigation.findNavController(this@StartActivity,... should be moved to a private method for a better readability (enterPasscode() ?)

Reply:
Calling it from onResume is too often. We need it once when the app is started  (onCreate) and then when it comes back from the background (activity count was zero)